packages/web3torrent/src/library/web3torrent-lib.ts

```diff
@@ -83,9 +88,13 @@ export default class WebTorrentPaidStreamingClient extends WebTorrent {
     });
     const raceResult = await Promise.race([gotAWire, timer]);
     if (torrentId) {
-      this.remove(torrentId);
+      super.remove(torrentId);
```

Comment:
If `resolve(torrentId)` replaces `resolve(true)`, we can avoid having to define the `torrentId` variable.

Reply:
Thing is, and correct me if I'm wrong, Promise.race only returns the first promise that gets solved, and drops the other. If the torrent is created but no peers connect on time (meaning, if the test fails), we won't have the torrentId.

Reply:
How about something like that?
```
    const gotAWire = new Promise<string>(resolve => {
      super.add(testTorrent.magnetURI, (torrent: Torrent) => {
        torrent.once('wire', () => resolve(torrent.infoHash));
      });
    });
    const timer = new Promise<void>(function(resolve, _) {
      setTimeout(resolve, timeOut);
    });
    const torrentId = await Promise.race([gotAWire, timer]);
    if (torrentId) {
      super.remove(torrentId);
    }
    this.clientCapability = torrentId
      ? TorrentClientCapabilities.CAPABLE
      : TorrentClientCapabilities.NOT_CAPABLE;
```

Reply:
but... doesn't that has the same problem? If timer resolves first, we lose the Promise result of `gotAWire`, and then we can't remove the torrent from the client

Reply:
Resolved on a call, thanks @spersico!